Backend/Models/Users/UserManager.cs

```diff
@@ -72,9 +72,22 @@ public static bool Authenticate(WebSocket socket, string username)
                 return false;
             }
 
-            user.Username = username;
+            User? userInDb = User.GetUserFromDB(username);
+            if (userInDb == null)
+            {
+                user.Username = username;
+                user.SaveToDb();
+            }
+            else
+            {
+                user = userInDb;
+            }
             user.IsRegistered = true;
```

Comment:
potentially move this below the isBanned check?

Reply:
will do